spec/features/users/sign_in_spec.rb

```diff
@@ -426,6 +426,38 @@
         end
       end
     end
+
+    context 'with sp' do
+      scenario 'redirects to home page with error' do
+        user = user_with_2fa
+        service_provider = ServiceProvider.find_by(issuer: OidcAuthHelper::OIDC_IAL1_ISSUER)
+        IdentityLinker.new(user, service_provider).link_identity(
+          verified_attributes: %w[openid email],
+        )
+
+        perform_in_browser(:one) do
+          visit_idp_from_sp_with_ial1(:oidc)
+          sign_in_live_with_2fa(user)
+
+          expect(current_url).to match('http://localhost:7654/auth/result')
+        end
+
+        perform_in_browser(:two) do
+          visit_idp_from_sp_with_ial1(:oidc)
+          sign_in_live_with_2fa(user)
+
+          expect(current_url).to match('http://localhost:7654/auth/result')
+        end
+
+        perform_in_browser(:one) do
+          visit account_path
+
+          expect(current_path).to eq new_user_session_path
+          expect(page).to have_content(t('devise.failure.session_limited'))
+          expect(page).to have_content(service_provider.friendly_name)
```

Comment:
There's a spec helper we can use for this:

https://github.com/18F/identity-idp/blob/51287d68629b9a6849d170c4010b65a382c95f17/spec/support/features/session_helper.rb#L710-L713

```suggestion
          expect_branded_experience
```